client.go

```diff
@@ -1830,3 +1836,33 @@ func flags(f int) uint32 {
 	}
 	return out
 }
+
+// posixPerm converts Go permission bits to POSIX permission bits,
+// masking off file type bits.
+//
+// This differs from fromFileMode in that it preserves the POSIX versions of
+// setuid, setgid and sticky in m, because we've historically supported those
+// bits.
+func posixPerm(m os.FileMode) uint32 {
+	// POSIX mode bits, as listed at
+	// https://tools.ietf.org/html/draft-ietf-secsh-filexfer-13#section-7.6
+	const (
+		posixSetuid = 04000
+		posixSetgid = 02000
+		posixSticky = 01000
+		posixBits   = 07777
+	)
+
+	posix := m & posixBits
+
+	switch {
```

Comment:
This switch means only one of these cases will be followed. If I pass in `m = os.ModeSetuid | os.ModeSetgid | os.ModeSticky` it will do the wrong thing: only apply `Setuid`.

Reply:
Fixed and tested.